### ISSUE TITLE
Update documentation

### ISSUE THREAD

**Issue by lasse-seivaag** [versionize/versionize#136]
First of all, thank you to all who have contributed here, it is a good project.

The readme (and other documentation) seems to be out of date / missing some important parts.

* We had issues with linking the CLI arguments to the .versionize configuration file
   *  --project-name does not state that it expects a value (--project-name <PROJECT_NAME>)
* Configuration file example is missing important properties which we needed to use
   *  TagTemplate is not mentioned at all in readme 

We had to dig through the source code to find the available options and correct usage.

**Comment by Barsan1** [comment]
> First of all, thank you to all who have contributed here, it is a good project.
> 
> The readme (and other documentation) seems to be out of date / missing some important parts.
> 
> * We had issues with linking the CLI arguments to the .versionize configuration file
>   
>   * --project-name does not state that it expects a value (--project-name <PROJECT_NAME>)
> * Configuration file example is missing important properties which we needed to use
>   
>   * TagTemplate is not mentioned at all in readme
>   * Could a full schema be part of the readme, or placed in source and linked to?
> 
> We had to dig through the source code to find the available options and correct usage.

The configuration file is just a Json file at the last commit i also mention this (ConfigurationContract.cs) #133
Also, i just replay to issue #137 and give an example take a look over there.

**Comment by lasse-seivaag** [comment]
@Barsan1 
Thank you for the reply! 

We did eventually find the options we were looking for (except for the use cases mentioned in #137 but I will read your suggestions there, maybe we missed something)

I created this issue to ask if the options in the code, could be / should be easily available in the README, or in a pre-filled template/schema file
(I see now that CommitParserOptions exists, maybe that is a solution to #137)

Feel free to close the issue if this is deemed unnecessary.

**Comment by RemiCELLARD** [comment]
> @Barsan1 Thank you for the reply!
> 
> We did eventually find the options we were looking for (except for the use cases mentioned in #137 but I will read your suggestions there, maybe we missed something)
> 
> I created this issue to ask if the options in the code, could be / should be easily available in the README, or in a pre-filled template/schema file (I see now that CommitParserOptions exists, maybe that is a solution to #137)
> 
> Feel free to close the issue if this is deemed unnecessary.

Hi,


That should be a great idea to have all these information about the `.versionize` file in the wiki tab of the github repo.
We can assume that we can add more information about each argument of the file. (random example : https://github.com/oobabooga/text-generation-webui/wiki/03-%E2%80%90-Parameters-Tab#parameters-description )

Because today, When I wanted to find out what parameters were proposed for the `.versionize` file, I didn't have the idea of looking directly at the `ConfigurationContract.cs` file (despite the fact that this is indicated in a small sentence at the end of the paragraph which is easily forgotten when reading).

What do you think ?

**Comment by cabauman** [comment]
Yeah, definitely. The stuff you're referring to (tag template, project-name, etc.) is part of the recently added mono-repo support, which hasn't been documented yet. My current plan is to update the docs as part of V2.

**Comment by Barsan1** [comment]
> Yeah, definitely. The stuff you're referring to (tag template, project-name, etc.) is part of the recently added mono-repo support, which hasn't been documented yet. My current plan is to update the docs as part of V2 (#135)

I can assist with that. Are you considering a new method of documentation, or simply updating the README file?

**Comment by cabauman** [comment]
Great! In the short term, just updating the readme. But if we eventually want to add some verbose guides, I'm open to utilizing the wiki, as an alternative to bloating the readme.

**Comment by Barsan1** [comment]
> Great! In the short term, I simply intend to update the readme. But if we eventually want to add some verbose guides (e.g. like how you assisted in that mono-repo issue), I'm open to utilizing the wiki, as an alternative to bloating the readme.

Utilizing the wiki option should be more robust, perhaps by using XML docs and then generating markdown files through a tool, followed by updates in the CI/CD pipeline. I notice you're working on V2, and I have some features that could contribute to this project. Updating the README manually is a tedious task, but it's sufficient for the moment. If you're considering using the wiki, I'd be interested in hearing your ideas.

Another approach could be to utilize a README and an examples folder with user stories similar to those in #137 
However, it all depends on the project's evolution and the strategies utilized to keep it up to date.

**Comment by RemiCELLARD** [comment]
Hi,

Here's a draft / sample for the wiki page : https://github.com/RemiCELLARD/versionize/wiki
I believe that this documentation work should be accompanied by the addition of documentation in the code itself.

Don't hesitate to give us feedback if you have other ideas for the wiki layout.

**Comment by cabauman** [comment]
@RemiCELLARD sorry, I'm just now noticing your wiki comment. That looks pretty nice, thanks for sharing. Let's wait and see if/when myself or another contributor will have time to build onto that POC.
